### PR TITLE
[RW-9991][risk=no] Added admin endpoint to delete persistent disks

### PR DESCRIPTION
**Pull request by evrii**
I manually tested by creating a runtime in a workspace as one user.
Then I got the user's disk from the response of calling GET {{api_url}}/v1/admin/workspaces/{{workspace}}/disks
Then as an admin user I ran the following endpoint on the corresponding disk: 
DELETE {{api_url}}/v1/admin/workspaces/{{workspace}}/disks/{{disk_name}}
I then ran GET {{api_url}}/v1/admin/workspaces/aou-rw-local1-a7d2f021/disks/all-of-us-pd-15-1f89 to confirm that the disk had been deleted.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 1]
```diff
@@ -672,20 +683,4 @@ public int stopAllUserAppsAsService(String userEmail) {
 
     return results.size();
   }
-
-  private List<LeonardoListPersistentDiskResponse> listPersistentDiskByProject(
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:676]: This is just a cleanup as it was not being used anymore correct?

File: api/src/test/java/org/pmiops/workbench/api/DiskAdminControllerTest.java [hunk 2]
```diff
@@ -83,4 +84,18 @@ public void listDisksInWorkspace_workspaceNotFound() {
     assertThrows(
         NotFoundException.class, () -> diskAdminController.listDisksInWorkspace(WORKSPACE_NS));
   }
+
+  @Test
+  public void deleteDisk() {
```
Comment (NehaBroad) [inline api/src/test/java/org/pmiops/workbench/api/DiskAdminControllerTest.java:89]: Not sure if we are doing this yet, should we have a test to check how service handles exception?

Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/api/DiskAdminControllerTest.java:89]: Ahh i now see a test listDisksInWorkspace_workspaceNotFound, we should add some negative test for deletedisk() or if leo threw an error while deleting the disk maybe unauthorized?

File: api/src/test/java/org/pmiops/workbench/disk/DiskServiceTest.java [hunk 3]
```diff
@@ -116,4 +117,25 @@ public void deleteDisk_workspaceNotFound() {
...
+  public void deleteDiskAsService() {
+    String diskName = user.generatePDName();
+    DbWorkspace workspace = new DbWorkspace().setGoogleProject(GOOGLE_PROJECT_ID);
+    when(mockWorkspaceService.lookupWorkspaceByNamespace(WORKSPACE_NS)).thenReturn(workspace);
+    diskService.deleteDiskAsService(WORKSPACE_NS, diskName);
+    verify(mockLeonardoApiClient).deletePersistentDiskAsService(GOOGLE_PROJECT_ID, diskName);
+  }
+
+  @Test
+  public void deleteDiskAsService_leoException() {
```
Comment (NehaBroad) [inline api/src/test/java/org/pmiops/workbench/disk/DiskServiceTest.java:131]: nit: notFoundException?

**Review by NehaBroad — APPROVED** [review]

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClient.java [hunk 4]
```diff
@@ -68,6 +68,10 @@ LeonardoGetPersistentDiskResponse getPersistentDisk(String googleProject, String
   /** Deletes a persistent disk */
   void deletePersistentDisk(String googleProject, String diskName) throws WorkbenchException;
 
+  /** Deletes a persistent disk as an admin */
+  void deletePersistentDiskAsService(String googleProject, String diskName)
```
[the remark this answers is not in the document]
Comment (evrii) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClient.java:72]: It seems as though most of our as admin functions use the "asService" pattern.

File: api/src/test/java/org/pmiops/workbench/api/DiskAdminControllerTest.java (hunk 2 above)
Comment (evrii) [reply api/src/test/java/org/pmiops/workbench/api/DiskAdminControllerTest.java:90]: Added tests

**Pull request merged**